deepchem/data/tests/test_shuffle.py

```diff
@@ -13,6 +9,47 @@
 import numpy as np
 
 
+def test_complete_shuffle_one_shard():
+  """Test that complete shuffle works with only one shard."""
+  X = np.random.rand(10, 10)
+  dataset = dc.data.DiskDataset.from_numpy(X)
+  shuffled = dataset.complete_shuffle()
+  assert len(shuffled) == len(dataset)
+  assert not np.array_equal(shuffled.ids, dataset.ids)
+  assert sorted(shuffled.ids) == sorted(dataset.ids)
+  assert shuffled.X.shape == dataset.X.shape
+  assert shuffled.y.shape == dataset.y.shape
+  assert shuffled.w.shape == dataset.w.shape
```

Comment:
On all these tests, we should check the full content of the arrays, not just their shapes.  Something like this:

```python
original_indices = dict((id, i) for i, id in enumerate(dataset.ids))
shuffled_indices = dict((id, i) for i, id in enumerate(shuffled.ids))
for id in dataset.ids:
  i = original_indices[id]
  j = shuffled_indices[id]
  assert np.array_equal(dataset.X[i], shuffled.X[j])
  assert np.array_equal(dataset.y[i], shuffled.y[j])
  assert np.array_equal(dataset.w[i], shuffled.w[j])
```

Reply:
Good point, will add in additional tests in this vein